File: MsSqlSemaphore/DotNetC6/MsSqlSemaphoreNc6/Helpers/DaoHelper.cs
﻿namespace pvWay.MsSqlSemaphore.nc6.Helpers;

internal class DaoHelper
{
    public static string TruncateThenEscape(string value, int maxLen)
    {
        if (string.IsNullOrEmpty(value)) return value;
        var val = value.Length <= maxLen
            ? value
            : value.Substring(0, maxLen - 3) + "...";
        return Escape(val);
    }

    private static string Escape(string value)
    {
        return value.Replace("'", "''");
    }

    /// <summary>
    /// returns a ready to concat sql string in
    /// the form 'yyyy-MM-dd HH:mm:ss.sss'
    /// </summary>
    /// <param name="utcNow"></param>
    /// <returns></returns>
    public static string GetTimestamp(DateTime utcNow) => $"'{utcNow:yyyy-MM-dd HH:mm:ss.sss}'";
    
}

File: MsSqlSemaphore/DocNetC8/PvWay.PgSqlSemaphoreService.nc8/Helpers/DaoHelper.py
class DaoHelper:
    @staticmethod
    def truncate_then_escape(value, max_len):
        if not value:
            return value
        val = value if len(value) <= max_len else value[:max_len - 3] + "..."
        return DaoHelper.escape(val)

    @staticmethod
    def escape(value):
        return value.replace("'", "''")

    # returns a ready to concat sql string in
    # the form 'yyyy-MM-dd HH:mm:ss.sss'
    
    @staticmethod
    def get_timestamp(utc_now):
        return f"'{utc_now:%Y-%m-%d %H:%M:%S.%f}'"[:-3]